src/ui/pages/ConnectionDetails/ConnectionDetails.tsx

```diff
@@ -138,12 +138,19 @@ const ConnectionDetails = () => {
 
   const verifyAction = () => {
     async function deleteConnection() {
+      if (!connectionDetails) return;
       await Agent.agent.connections.deleteConnectionById(
         connectionShortDetails.id
       );
-      const updatedConnections = connectionsData.filter(
-        (item) => item.id !== connectionDetails?.id
+
+      const newConnectionsData = Object.fromEntries(
+        Object.entries(connectionsData).filter(
+          ([key]) => key !== connectionDetails?.id
```

Comment:
Is `connectionDetails.id` the same as `connectionShortDetails.id`? If so, I think we can just use that and remove your line `if (!connectionDetails) return;`. Agree @sdisalvo-crd ?

Reply:
@iFergal Looking at what was done above with `setLoading` I think he wanted to wait for `connectionDetails` to be there as a way to be sure that all the information has been fetched before performing any action, but yeah, if all we need here is the `id` and nothing else we can definitely use the one we already have in the `connectionShortDetails`. 

Also, if `loading` is true we shouldn't be able to do anything in theory...

Reply:
yep, shouldn't happen from a UI perspective and if it does, we have the info in `connectionShortDetails` - great, thanks :)